test/helpers/DaoBuilder.sol

```diff
@@ -0,0 +1,299 @@
+                )
+            );
+        }
+
+        // Emergency Multisig
+        {
+            EmergencyMultisig.MultisigSettings memory settings = EmergencyMultisig.MultisigSettings({
+                onlyListed: onlyListed,
+                minApprovals: minApprovals,
+                addresslistSource: multisig
+            });
+
+            emergencyMultisig = EmergencyMultisig(
+                createProxyAndCall(
+                    address(EMERGENCY_MULTISIG_BASE), abi.encodeCall(EmergencyMultisig.initialize, (dao, settings))
+                )
+            );
+        }
+
+        // Optimistic can execute on the DAO
+        dao.grant(address(dao), address(optimisticPlugin), dao.EXECUTE_PERMISSION_ID());
+
+        // Multisig can create proposals on the optimistic plugin
+        dao.grant(address(optimisticPlugin), address(multisig), optimisticPlugin.PROPOSER_PERMISSION_ID());
+
+        // Emergency Multisig can create proposals on the optimistic plugin
+        dao.grant(address(optimisticPlugin), address(emergencyMultisig), optimisticPlugin.PROPOSER_PERMISSION_ID());
+
+        if (optimisticProposers.length > 0) {
+            for (uint256 i = 0; i < optimisticProposers.length; i++) {
+                dao.grant(address(optimisticPlugin), optimisticProposers[i], optimisticPlugin.PROPOSER_PERMISSION_ID());
+            }
+        } else {
+            // Ensure that at least the owner can propose
+            dao.grant(address(optimisticPlugin), owner, optimisticPlugin.PROPOSER_PERMISSION_ID());
+        }
+
+        // Revoke transfer ownership to the owner
+        dao.grant(address(dao), owner, dao.ROOT_PERMISSION_ID());
+        dao.revoke(address(dao), address(this), dao.ROOT_PERMISSION_ID());
+
+        // Labels
+        vm.label(address(dao), "DAO");
+        vm.label(address(optimisticPlugin), "OptimisticPlugin");
+        vm.label(address(votingToken), "VotingToken");
+        vm.label(address(multisig), "Multisig");
+        vm.label(address(emergencyMultisig), "EmergencyMultisig");
+        vm.label(address(taikoL1), "TaikoL1");
+        vm.label(address(taikoBridge), "TaikoBridge");
+
+        // Moving forward to avoid proposal creations failing or getVotes() giving inconsistent values
+        blockForward(1);
+        timeForward(1);
+    }
+
+    /// @notice Moves the EVM time forward by a given amount.
+    /// @param time The amount of seconds to advance.
+    function timeForward(uint256 time) internal {
+        vm.warp(block.timestamp + time);
+    }
+
+    /// @notice Moves the EVM time back by a given amount.
+    /// @param time The amount of seconds to subtract.
+    function timeBack(uint256 time) internal {
+        vm.warp(block.timestamp - time);
+    }
+
+    /// @notice Sets the EVM timestamp.
+    /// @param timestamp The timestamp in seconds.
+    function setTime(uint256 timestamp) internal {
```

Comment:
In term of review clarity, I keep finding `vm.warp()` and `vm.roll()` very confusing. Even after using them hundreds of times. Happy to remove them if you insist but the fact that we can add logic on top and make things even more readable makes a big difference to me.

Reply:
I understand and feel you. But I think it's better to use them as they come, otherwise newcomers into the codebase will struggle. 
Yeah, please, let change them :)